src/auto/AutoCriterion.lua

```diff
@@ -0,0 +1,26 @@
+local AutoCriterion, parent = torch.class('nn.AutoCriterion', 'nn.Criterion')
```

Comment:
Done for maintainability, though require 'nn' is implicitely required in autograd.nnwrapper above.